### PR TITLE
Refactor morgues and crematorium to use overlays and sensible names, add wirecutter act on morgue to make it consistent with bodybags

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR Do
This reworks the sprites for the morgue and crematorium, relying more on overlays, and also giving them less confusing names in the process. To go in line with the names, they're also used in code instead of the more cryptic list ordering with mismatched numbered sprites. Trays are also below the object layer now, as they'd inconsistently be displayed below or above objects otherwise.

At the same time, this also adds the removal of tags with wirecutters to be consistent with bodybags, just in case you happen to lose your pen.

## Why It's Good For The Game
morgue1, morgue2, morgue3... is bad. Smaller sprites too.

## Images of changes
Morgue Crematorium
Morgue
Tray_Bodybags

## Changelog
:cl:
add: Removal of tags on morgues can be done with wirecutters as well.
fix: Objects are displayed on top of morgue and crematorium trays consistently.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -14,18 +14,18 @@
...
 	dir = EAST
 	var/obj/structure/m_tray/connected = null
 	var/list/status_descriptors = list(
-	"The tray is currently extended.",
-	"The tray is currently empty.",
-	"The tray contains an unviable body.",
-	"The tray contains a body that is responsive to revival techniques.",
-	"The tray contains something that is not a body.",
-	"The tray contains a body that might be responsive."
+	"extended" = "The tray is currently extended.",
```
Comment (farie82) [inline code/game/objects/structures/morgue.dm:23]: Could you make defines out of these?

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -35,41 +35,35 @@
...
-				desc = initial(desc) + "\n[status_descriptors[5]]"
+				status = "notbody"
 		else
-			icon_state = "morgue1"
-			desc = initial(desc) + "\n[status_descriptors[2]]"
+			status = "empty"
+		add_overlay("morgue_[status]")
+	else
+		status = "extended"
+	desc = initial(desc) + "\n[status_descriptors[status]]"
```
Comment (farie82) [inline code/game/objects/structures/morgue.dm:64]: I'd say leave the `desc` alone and instead override the `examine` proc to include the status description

File: code/game/objects/structures/morgue.dm [hunk 3]
```diff
@@ -128,29 +110,35 @@
 		var/t = rename_interactive(user, P)
 		if(isnull(t))
 			return
-		cut_overlays()
-		if(t)
-			add_overlay(image(icon, "morgue_label"))
+		update()
 		add_fingerprint(user)
 		return
+	if(istype(P, /obj/item/wirecutters) && name != initial(name))
```
Comment (farie82) [inline code/game/objects/structures/morgue.dm:116]: Please use `wirecutter_act` instead

File: code/game/objects/structures/morgue.dm [hunk 4]
```diff
@@ -128,29 +110,35 @@
...
+		name = initial(name)
+		update()
+		return
 	return ..()
 
 /obj/structure/morgue/relaymove(mob/user as mob)
 	if(user.stat)
 		return
+	connect()
+	return
```
Comment (farie82) [inline code/game/objects/structures/morgue.dm:127]:
```suggestion
```

File: code/game/objects/structures/morgue.dm [hunk 5]
```diff
@@ -256,14 +244,19 @@
...
+		add_overlay("crema_closed")
+		if(cremating)
+			add_overlay("crema_active")
+			return
 		if(contents.len)
-			icon_state = "crema2"
-		else
-			icon_state = "crema1"
+			add_overlay("crema_full")
 	return
```
Comment (farie82) [inline code/game/objects/structures/morgue.dm:260]:
```suggestion
```

File: code/game/objects/structures/morgue.dm [hunk 6]
```diff
@@ -327,19 +309,22 @@
 /obj/structure/crematorium/relaymove(mob/user as mob)
 	if(user.stat || locked)
 		return
+	connect()
+	return
```
Comment (farie82) [inline code/game/objects/structures/morgue.dm:313]:
```suggestion
```
No need to place a `return` at the end of a proc if it returns nothing.

**Review by farie82 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/morgue.dm (hunk 1 above)
Comment (Vi3trice) [reply code/game/objects/structures/morgue.dm:23]: I couldn't really figure out how to do it while making it work with the overlay names at the same time. If they were only used for descriptions, then yes defines would have been better.

**Review by farie82 — APPROVED** [review]
LGTM!

File: code/game/objects/structures/morgue.dm [hunk 7]
```diff
@@ -10,68 +10,70 @@
...
 	name = "morgue"
 	desc = "Used to keep bodies in until someone fetches them."
 	icon = 'icons/obj/stationobjs.dmi'
-	icon_state = "morgue1"
+	icon_state = "morgue"
 	density = 1
 	max_integrity = 400
 	dir = EAST
 	var/obj/structure/m_tray/connected = null
 	var/list/status_descriptors = list(
```
Comment (dearmochi) [inline code/game/objects/structures/morgue.dm:29]: This variable could be `static` or outright axed to save up on memory.

**Review by dearmochi — CHANGES_REQUESTED** [review]

**Review by dearmochi — APPROVED** [review]

**Pull request merged**